kernel/stack.c

```diff
@@ -85,19 +86,19 @@ Z_SYSCALL_HANDLER(k_stack_alloc_init, stack, num_entries)
 
 void k_stack_cleanup(struct k_stack *stack)
 {
-	__ASSERT_NO_MSG(!_waitq_head(&stack->wait_q));
+	__ASSERT_NO_MSG(_waitq_head(&stack->wait_q) == NULL);
 
-	if (stack->flags & K_STACK_FLAG_ALLOC) {
+	if ((stack->flags & K_STACK_FLAG_ALLOC) == K_STACK_FLAG_ALLOC) {
 		k_free(stack->base);
 		stack->base = NULL;
-		stack->flags &= ~K_STACK_FLAG_ALLOC;
+		stack->flags &= ~(u8_t)K_STACK_FLAG_ALLOC;
```

Comment:
Put this cast into the definition of K_STACK_FLAG_ALLOC, not here.  If the flag needs to fit in a byte, it needs to be defined that way.

Reply:
Did this change in the definition.